src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java

```diff
@@ -40,20 +41,24 @@ public ValidationOutputData addExperimentToDB(KruizeExperimentEntry kruizeExperi
         Transaction tx = null;
         String statusValue = "failure";
         Timer.Sample timerAddExpDB = Timer.start(MetricsConfig.meterRegistry());
-        try (Session session = KruizeHibernateUtil.getSessionFactory().openSession()) {
-            try {
-                tx = session.beginTransaction();
-                session.persist(kruizeExperimentEntry);
-                tx.commit();
-                validationOutputData.setSuccess(true);
-                statusValue = "success";
-            } catch (HibernateException e) {
-                LOGGER.error("Not able to save experiment due to {}", e.getMessage());
-                if (tx != null) tx.rollback();
-                e.printStackTrace();
-                validationOutputData.setSuccess(false);
-                validationOutputData.setMessage(e.getMessage());
-                //todo save error to API_ERROR_LOG
+        try {
+            addPartitions(DBConstants.TABLE_NAMES.KRUIZE_RESULTS);
```

Comment:
pass year and month as a parameter from here so that add Partitions loosely coupled .  and also parameter that says if 
partitions by days or month or for every 15 days etcs

Reply:
Updated as discussed